From: Simon McVittie <smcv@debian.org>
Date: Sun, 6 Sep 2020 20:00:58 +0100
Subject: Link pipewire-jack to libatomic if required

This is necessary on some architectures that implement atomic operations
as library calls, including Debian's armel and mipsel ports.

Forwarded: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/299
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 meson.build                   | 13 +++++++++++++
 pipewire-jack/src/meson.build |  2 +-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index adabccc..d43023c 100644
--- a/meson.build
+++ b/meson.build
@@ -155,6 +155,19 @@ elif cc.has_argument('-mfpu=neon')
   endif
 endif
 
+if cc.links('''
+  #include <stdint.h>
+
+  int main(void) {
+    int64_t eight;
+    __atomic_store_n(&eight, 123, __ATOMIC_SEQ_CST);
+    return 0;
+  }
+  ''', name : '8-byte __atomic_store_n without libatomic')
+  atomic_dep = dependency('', required: false)
+elif cc.has_function('__atomic_store_8', args: '-latomic')
+  atomic_dep = cc.find_library('atomic')
+endif
 
 cdata = configuration_data()
 cdata.set('PIPEWIRE_VERSION_MAJOR', pipewire_version_major)
diff --git a/pipewire-jack/src/meson.build b/pipewire-jack/src/meson.build
index a052eaa..0795795 100644
--- a/pipewire-jack/src/meson.build
+++ b/pipewire-jack/src/meson.build
@@ -39,7 +39,7 @@ pipewire_jack = shared_library('jack',
     version : libversion,
     c_args : pipewire_jack_c_args,
     include_directories : [configinc],
-    dependencies : [pipewire_dep, jack_dep, mathlib],
+    dependencies : [pipewire_dep, atomic_dep, jack_dep, mathlib],
     install : true,
     install_dir : libjack_path,
 )
